src/modules/esl-utils/misc/array.ts

```diff
@@ -2,6 +2,21 @@ import {identity} from './functions';
 
 type Tuple<T> = [T?, T?];
 
+interface ArrayLike<T = any> {
+  [key: number]: T;
+  length: number;
+}
+
+const isArrayLike = (value: any): value is ArrayLike => {
```

Comment:
I think we can also export this utility to outside world

Reply:
That's a great idea!
But I think a more appropriate location would be an object so it is still a check that the object is a pseudo-array.
Also, I think we need to check that the object is iterable, i.e. we need a method to check - isIterable() and call it inside method unwrap(). Although I'm not sure about it. :confused:

Reply:
Great addition.
I'm ok with iterables in case we still will be compatible with IE, but it is a low priority.
Also regarding our current main case, I think we should not unwrap string.
Basically our real cases for now:
 - array or element - so situation, when we are not sure, is the input will be an array or item, with the assumption that the item will not be array-like
 - NodeList or Node - so we unwrap NodeList but not a node 
 - JQuery collection or HTML Element - again we are not sure about the input but want to have the JQuery collection unwrapped (we can not include even types of JQ to the library for sure, but it will be nice to have that case covered)